Pearson-Advance/edx-platform
# PADV 1025 - Notes tab does not display notes for ccx

## Ticket

https://agile-jira.pearson.com/browse/PADV-1025

## Description

As 1025 describes, the notes for a CCX are failing to show in the Notes tab. The problem is caused because when the note is saved, the course_id value is passed containing branch and version, causing the Notes tab to not find any notes since it is searching with the CCX course_id without branch nor version. This PR applies a **hotfix** for the issue, the problem is more complex and requires further review to conclude if this is the best solution.

## Changes

- Change edx notes saving function to use not versioned course id.

## How to test?

- Setup devstack.
- Setup edx notes api, here are the steps:

1. uncomment these lines from your devstack https://github.com/Pearson-Advance/devstack/blob/pearson-release/olive.main/docker-compose.yml#L384-L411 and https://github.com/Pearson-Advance/devstack/blob/pearson-release/olive.main/docker-compose-host.yml#L34-L37
2. execute make dev.up.edx_notes_api
3. In edx notes go to notesserver/settings/devstack.py and configure the values to match your devstack
4. attach a shell to the edx notes container and cd to /edx/apps/edx_notes_api/edx_notes_api
5. Create a virtualenv and execute make create-index

- Create a course with CCX enabled.
- Enable notes (go to advance settings and search for "Enable Student Notes")
- Create a CCX
- Create a note in the CCX (https://edx.readthedocs.io/projects/open-edx-learner-guide/en/latest/SFD_notes.html#overview-of-notes)
- Verify that the note created is being showed in the Notes tab
